File: app/src/main/java/com/bq/zowi/rx/RetryWithDelay.java
package com.bq.zowi.rx;

import com.bq.zowi.utils.Grove;
import java.util.concurrent.TimeUnit;
import rx.Observable;
import rx.functions.Func1;

/* JADX INFO: loaded from: classes.dex */
public class RetryWithDelay implements Func1<Observable<? extends Throwable>, Observable<?>> {
    private final int maxRetries;
    private int retryCount = 0;
    private final int retryDelayMillis;

    static /* synthetic */ int access$004(RetryWithDelay x0) {
        int i = x0.retryCount + 1;
        x0.retryCount = i;
        return i;
    }

    public RetryWithDelay(int maxRetries, int retryDelayMillis) {
        this.maxRetries = maxRetries;
        this.retryDelayMillis = retryDelayMillis;
    }

    @Override // rx.functions.Func1
    public Observable<?> call(Observable<? extends Throwable> attempts) {
        return attempts.flatMap(new Func1<Throwable, Observable<?>>() { // from class: com.bq.zowi.rx.RetryWithDelay.1
            @Override // rx.functions.Func1
            public Observable<?> call(Throwable throwable) {
                if (RetryWithDelay.access$004(RetryWithDelay.this) < RetryWithDelay.this.maxRetries) {
                    Grove.d(throwable, "Rx Retry: " + RetryWithDelay.this.retryCount, new Object[0]);
                    return Observable.timer(RetryWithDelay.this.retryDelayMillis, TimeUnit.MILLISECONDS);
                }
                Grove.d(throwable, "Rx Max retries hit. Passing error along", new Object[0]);
                return Observable.error(throwable);
            }
        });
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/rx/RetryWithDelay.kt
package com.bq.zowi.rx

import com.bq.zowi.utils.Grove
import io.reactivex.Observable
import io.reactivex.functions.Function
import java.util.concurrent.TimeUnit

class RetryWithDelay(
    private val maxRetries: Int,
    private val retryDelayMillis: Int
) : Function<Observable<out Throwable>, Observable<*>> {

    private var retryCount = 0

    override fun apply(attempts: Observable<out Throwable>): Observable<*> {
        return attempts.flatMap { throwable ->
            if (retryCount++ < maxRetries) {
                Grove.d(throwable, "Rx Retry: $retryCount")
                Observable.timer(retryDelayMillis.toLong(), TimeUnit.MILLISECONDS)
            } else {
                Grove.d(throwable, "Rx Max retries hit. Passing error along")
                Observable.error<Any>(throwable)
            }
        }
    }
}
